pkg/kubectl/cmd/edit.go

```diff
@@ -210,7 +210,7 @@ outter:
 				// which means our changes are invalid or (2) when we exit the second
 				// time. The second case is more usual so we can probably live with it.
 				// TODO: A less hacky fix would be welcome :)
-				fmt.Fprintln(out, "Edit cancelled, no valid changes were saved.")
```

Comment:
Sorry, I wasn't clear.  We should be passing stderr down into the command, and each of these should be errOut or something similar.  We shouldn't ever write to stderr directly inside a command or below.